File: Write-File/app/src/main/java/com/example/writefile/SavedFiles/FileAdapter.java
package com.example.writefile.SavedFiles;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.writefile.R;

import java.util.List;

public class FileAdapter extends RecyclerView.Adapter<FileAdapter.ContentViewHolder> {
    private OnRVItemClickListener clickListener;
    private List<FilePojo> fileList;
    public FileAdapter(List<FilePojo> fileList, OnRVItemClickListener clickListener) {
        this.fileList = fileList;
        this.clickListener = clickListener;
    }

    @NonNull
    @Override
    public ContentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view  = inflater.inflate(R.layout.list_item,parent,false);
        return new ContentViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ContentViewHolder holder, int position) {
        String filename = fileList.get(position).getFilename();
        String filesize = fileList.get(position).getFilesize();

        holder.fname.setText(filename);
        holder.fsize.setText(filesize);
    }

    @Override
    public int getItemCount() {
        return fileList.size();
    }

    public class ContentViewHolder extends RecyclerView.ViewHolder{
        TextView fname,fsize;
        ImageView load,share;
        public ContentViewHolder(@NonNull View itemView) {
            super(itemView);
            fname = itemView.findViewById(R.id.fileName);
            fsize = itemView.findViewById(R.id.fileSize);
            load = itemView.findViewById(R.id.loadFile);
            share = itemView.findViewById(R.id.shareFile);
            load.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    clickListener.onClick(getAdapterPosition());
                }
            });
            share.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    clickListener.onShare(getAdapterPosition());
                }
            });
        }
    }

}

File: Kotlin-File-Writer/app/src/main/java/ca/nuro/nuos/kotlinfilewriter/supports/FileAdapter.kt
package ca.nuro.nuos.kotlinfilewriter.supports


import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ca.nuro.nuos.kotlinfilewriter.R
import ca.nuro.nuos.kotlinfilewriter.dataobjects.FileObj

class FileAdapter(private val respArray: ArrayList<FileObj>) :
    RecyclerView.Adapter<FileAdapter.ContentViewHolder>() {

    private var onItemClickListener: OnItemClickListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContentViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.file, parent, false)
        return ContentViewHolder(itemView, onItemClickListener!!)
    }

    override fun onBindViewHolder(holder: ContentViewHolder, position: Int) {
        holder.fileName.text = respArray[position].fileName
        holder.fileSize.text = respArray[position].fileSize
    }

    override fun getItemCount(): Int {
        return respArray.size
    }

    inner class ContentViewHolder(itemView: View, listener: OnItemClickListener) :
        RecyclerView.ViewHolder(itemView) {
        val fileName: TextView = itemView.findViewById(R.id.fileName)
        val fileSize: TextView = itemView.findViewById(R.id.fileSize)
        private val share:ImageView = itemView.findViewById(R.id.shareFile)
        private val load:ImageView = itemView.findViewById(R.id.loadFile)
        init {
            val position: Int = adapterPosition
            share.setOnClickListener {
                if (position != RecyclerView.NO_POSITION) {
                    listener.onShare(adapterPosition)
                }
            }
            load.setOnClickListener {
                if (position != RecyclerView.NO_POSITION) {
                    listener.onLoad(adapterPosition)
                }
            }
        }
    }

    fun setOnItemShareListener(listener: OnItemClickListener) {
        onItemClickListener = listener
    }


}